packages/dd-trace/test/dogstatsd.spec.js

```diff
@@ -376,61 +377,76 @@ describe('dogstatsd', () => {
     it('.increment()', () => {
       client = new CustomMetrics({ dogstatsd: {} })
 
+      client.increment('test.count', 10)
       client.increment('test.count', 10)
       client.flush()
 
       expect(udp4.send).to.have.been.called
-      expect(udp4.send.firstCall.args[0].toString()).to.equal('test.count:10|c\n')
+      expect(udp4.send.firstCall.args[0].toString()).to.equal('test.count:20|g\n')
     })
 
     it('.increment() with default', () => {
       client = new CustomMetrics({ dogstatsd: {} })
 
+      client.increment('test.count')
       client.increment('test.count')
       client.flush()
 
       expect(udp4.send).to.have.been.called
-      expect(udp4.send.firstCall.args[0].toString()).to.equal('test.count:1|c\n')
+      expect(udp4.send.firstCall.args[0].toString()).to.equal('test.count:2|g\n')
     })
 
     it('.decrement()', () => {
       client = new CustomMetrics({ dogstatsd: {} })
 
+      client.decrement('test.count', 10)
       client.decrement('test.count', 10)
       client.flush()
 
       expect(udp4.send).to.have.been.called
-      expect(udp4.send.firstCall.args[0].toString()).to.equal('test.count:-10|c\n')
+      expect(udp4.send.firstCall.args[0].toString()).to.equal('test.count:-20|g\n')
     })
 
     it('.decrement() with default', () => {
       client = new CustomMetrics({ dogstatsd: {} })
 
+      client.decrement('test.count')
       client.decrement('test.count')
       client.flush()
 
       expect(udp4.send).to.have.been.called
-      expect(udp4.send.firstCall.args[0].toString()).to.equal('test.count:-1|c\n')
+      expect(udp4.send.firstCall.args[0].toString()).to.equal('test.count:-2|g\n')
     })
 
     it('.distribution()', () => {
       client = new CustomMetrics({ dogstatsd: {} })
 
+      client.distribution('test.dist', 10)
       client.distribution('test.dist', 10)
       client.flush()
 
       expect(udp4.send).to.have.been.called
-      expect(udp4.send.firstCall.args[0].toString()).to.equal('test.dist:10|d\n')
+      expect(udp4.send.firstCall.args[0].toString()).to.equal('test.dist:10|d\ntest.dist:10|d\n')
     })
 
     it('.histogram()', () => {
       client = new CustomMetrics({ dogstatsd: {} })
 
+      client.histogram('test.histogram', 10)
       client.histogram('test.histogram', 10)
       client.flush()
 
       expect(udp4.send).to.have.been.called
-      expect(udp4.send.firstCall.args[0].toString()).to.equal('test.histogram:10|h\n')
+      expect(udp4.send.firstCall.args[0].toString()).to.equal([
+        'test.histogram.min:10|g',
+        'test.histogram.max:10|g',
+        'test.histogram.sum:20|c',
+        'test.histogram.total:20|c',
+        'test.histogram.avg:10|g',
+        'test.histogram.count:2|c',
+        'test.histogram.median:10.074696689511441|g',
```

Comment:
Is this a bug in the histogram calculation?

Reply:
My understanding is that this is normal as the histogram doesn't store values directly, it stores approximations, so while the values are expected to be close to the originals, there is no guarantee of being exact.